app/jobs/run_campaign_job.rb

```diff
@@ -0,0 +1,32 @@
+class RunCampaignJob < ApplicationJob
+  queue_as :critical
+
+  # Job that decides users the be matched for a given campaign at a given point in time.
+  # This job creates the matches.
+
+  def perform(campaign)
+    Rails.logger.info("Run RunCampaignJob for campaign_id #{campaign.id}")
+    return unless campaign.matching_algo_v2?
+    return unless campaign.running?
+    return campaign.completed! if campaign.remaining_doses <= 0
+    return campaign.completed! if Time.now.utc >= campaign.ends_at
+
+    # compute how many more users we need to match
+    limit = campaign.target_matches_count - campaign.matches.pending.alive.count
```

Comment:
We could also remove matches with email clicks + ~10 min from pending matches, to give users the time to reach the info. If they do not confirm they are probably not interested, and this will allow us to have a more dynamic feedback loop

Reply:
Attention avec ça, beaucoup d'anti-spam/antivirus professionnels, et pas seulement, cliquent sur les liens dans les emails pour les analyser. Donc un clic sur le lien d'un mail c'est pas forcément fiable.

Reply:
Gardons cette idee pour un deuxième temps, je ship comme ca